algosdk/future/transaction.py

```diff
@@ -3010,31 +3011,51 @@ def assign_group_id(txns, address=None):
     return result
 
 
-def wait_for_confirmation(algod_client, txid, wait_rounds=0, **kwargs):
+def wait_for_confirmation(
+    algod_client: algod.AlgodClient, txid: str, wait_rounds: int = 0, **kwargs
```

Comment:
Set the default value here instead of checking for 0 and overriding it.

A smaller default like 5 would probably be better default than 1000.

Reply:
This is my attempt at maintaining backwards compatibility, since we already shipped a version with this function (unlike the other SDKs).

There might be existing code that passes 0 for `wait_rounds` and expects the function to wait indefinitely, so I though 1000 was the best replacement for it.